src/Plugin/Action/BlockIP.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\Action\BlockIP.
+ */
+
+namespace Drupal\rules\Plugin\Action;
+
+use Drupal\ban\BanIpManagerInterface;
+use Drupal\Core\Annotation\Action;
+use Drupal\Core\Annotation\Translation;
+use Drupal\Core\Annotation\ContextDefinition;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\rules\Core\RulesActionBase;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides the 'Block IP' action.
+ *
+ * @Action(
+ *   id = "rules_block_ip",
+ *   label = @Translation("Block IP"),
+ *   category = @Translation("System"),
+ *   context = {
+ *     "ip" = @ContextDefinition("string",
+ *       label = @Translation("IP Address"),
+ *       description = @Translation("Block an IP using the Ban Module. If no IP is provided, the current user IP is used."),
+ *       required = false
+ *     )
+ *   }
+ * )
+ *
+ * @todo: Add access callback information from Drupal 7.
+ * @todo: We should maybe use a dedicated data type for the ip address, as we
+ * do in Drupal 7.
+ * @todo: This action depends on the ban module. We need to have a way to
+ * specify this.
+ */
+class BlockIP extends RulesActionBase implements ContainerFactoryPluginInterface {
```

Comment:
So this is in the system category so the class and file should be prefixed with "System". Same for the test class.